txpool/txpool.go

```diff
@@ -346,63 +345,48 @@ func (p *TxPool) Pop(tx *types.Transaction) {
 	}
 }
 
-// Drop removes the given (unrecoverable) transaction from
-// its associated promoted queue (account) and rolls
-// back the account's nextNonce.
-// Will update executables with the next primary
-// from that account (if any).
+// Drop clears the entire account associated with the given transaction
+// and reverts its next (expected) nonce.
 func (p *TxPool) Drop(tx *types.Transaction) {
+	// drop the entire account and revert nonce
```

Comment:
comment is a bit misleading as the line below it doesn't do the dropping

Reply:
[Removed](https://github.com/0xPolygon/polygon-edge/pull/380#:~:text=fix%20Drop()%20method-,7a9d85a,-Merge%20state)